src/chapters/typescript/functions.rst

```diff
@@ -0,0 +1,139 @@
+
+Anonymous Functions
+^^^^^^^^^^^^^^^^^^^
+
+For an anonymous function, you still need to provide types for the value returned and the parameters.
+
+.. admonition:: Example
+
+   Now we want to declare an anonymous function, ``myFunction``, which has one parameter, a number called ``x``, and returns the value of ``x`` multiplied by 2.
+
+   .. sourcecode:: typescript
+
+      let myFunction = function(x: number): number { return x*2; };
+
+Just as we did above with the named function, we need to make sure that we include the data type for the parameters and the return type of the function.
+
+Optional Parameter
+------------------
+
+In JavaScript, you can declare a function with 5 parameters and only give it 2 when it comes time to use it.
+In TypeScript, that is not the case.
+In TypeScript when you declare a function with 5 parameters, you have to give the function 5 arguments when calling it, unless you make some of those parameters optional.
+As with JavaScript functions, optional parameters can be omitted.
+This means that when you are calling the function, you can leave off the optional parameter(s).
+
+To denote a parameter as optional, you can use ``?`` notation. Any parameters that are optional must follow the required parameters.
+
+.. admonition:: Example
+
+   .. sourcecode:: typescript
+
+      function myFunction(a: number, b?:number): number {
+         return a+b+5;
+      }
+
+      console.log(myFunction(1,2));
+      console.log(myFunction(1));
+      console.log(myFunction(3,5));
+      console.log(myFunction(3));
+
+   ::
+
+      8
+      6
+      13
+      8
+
+In this example, the ``myFunction`` function has two parameters, ``a`` and ``b``.
+``a`` is required and ``b`` is optional.
+When only one argument is provided to the function, then the value of ``a+5`` is returned.
+When both arguments are provided, then the sum of the 2 arguments and 5 is returned.
+
+Another way to deal with a parameter you need to be optional is to give it a default value.
+
+Let's say that in the example above, we wanted to give ``b`` a default value of 5.
+That way, if no argument is supplied for ``b``, then the value returned is that of ``a+10``.
+
+.. admonition:: Example
+
+   .. sourcecode:: typescript
+
+      function myFunction(a: number, b = 5): number {
+         return a+b+5;
+      }
+
+      console.log(myFunction(1,2));
+      console.log(myFunction(1));
+      console.log(myFunction(3,6));
+      console.log(myFunction(3));      
+
+   :: 
+
+      8
+      11
+      14
+      13
+
+Because ``b`` has a default value of ``5``, when the user does not pass a value to the second argument of ``myFunction``, 5 is used as the value of ``b``.
+
+Check Your Understanding
+------------------------
+
+.. admonition:: Question
+
+   What is wrong with this function declaration? NOTE there are 3 things that are wrong here.
+
+   .. sourcecode:: typescript
+
+      let myFunction = function(a:string,b??:number) {return a*b};
```

Comment:
we should also add this concept check question to the canvas quizzes!

Reply:
Yes!